language/movie_nl/main.py

```diff
@@ -0,0 +1,342 @@
+            'type': 'PLAIN_TEXT',
+            'content': text,
+        },
+        'features': {
+            'extract_syntax': syntax,
+            'extract_entities': entities,
+            'extract_document_sentiment': sentiment,
+        },
+        'encoding_type': 'UTF32'
+    }
+
+    return body
+
+
+def get_sentiment(service, sentence):
+    """Get the sentence-level sentiment."""
+    body = get_request_body(
+        sentence, syntax=False, entities=True, sentiment=True)
+
+    docs = service.documents()
+    request = docs.annotateText(body=body)
+
+    response = request.execute(num_retries=3)
+
+    sentiment = response.get('documentSentiment')
+
+    if sentiment is None:
+        return (None, None)
+    else:
+        pol = sentiment.get('polarity')
+        mag = sentiment.get('magnitude')
+
+    if pol is None and mag is not None:
+        pol = 0
+    return (pol, mag)
+
+
+class Document(object):
+    """Document class captures a single document of movie reviews."""
+
+    def __init__(self, text, doc_id, doc_path):
+        self.text = text
+        self.doc_id = doc_id
+        self.doc_path = doc_path
+        self.sentence_entity_pair = None
+        self.label = None
+
+    def extract_all_sentences(self, service):
+        """Extract the sentences in a document."""
+
+        if self.sentence_entity_pair is not None:
+            return self.sentence_entity_pair
+
+        docs = service.documents()
+        request_body = get_request_body(
+            self.text,
+            syntax=True,
+            entities=True,
+            sentiment=False)
+        request = docs.annotateText(body=request_body)
+
+        ent_list = []
+
+        response = request.execute()
+        entities = response.get('entities', [])
+        sentences = response.get('sentences', [])
+
+        sent_list = [
+            sentence.get('text', {}).get('content') for sentence in sentences
+        ]
+
+        for entity in entities:
+            ent_type = entity.get('type')
+            wiki_url = entity.get('metadata', {}).get('wikipedia_url')
+
+            if ent_type == 'PERSON' and wiki_url is not None:
+                ent_list.append(wiki_url)
+
+        self.sentence_entity_pair = (sent_list, ent_list)
+
+        return self.sentence_entity_pair
+
+
+def to_sentiment_json(doc_id, sent, label):
+    """Convert the sentiment info to json."""
+    json_doc = {}
+
+    json_doc['doc_id'] = doc_id
+    json_doc['sentiment'] = float('%.3f' % sent)
+    json_doc['label'] = label
+
+    return json.dumps(json_doc)
+
+
+def get_wiki_title(wiki_url):
+    """Get the wikipedia page title for a given wikipedia URL."""
+    try:
+        content = requests.get(wiki_url).text
+        return content.split('title')[1].split('-')[0].split('>')[1].strip()
+    except:
+        return os.path.basename(wiki_url).replace('_', ' ')
+
+
+def to_entity_json(entity, entity_sentiment):
+    """Convert entities and their associated sentiment to json."""
```

Comment:
This is one place I think doc strings will help.

As Jon says I think for self-explanatory string types or int types, we can skip doc strings, otherwise we should have them